templates/webapi/admin/needle/index.html.ep

```diff
@@ -81,7 +92,17 @@
                         <option value="max14">Not last 2 weeks</option>
                         <option value="max28">Not last 4 weeks</option>
                         <option value="max61">Not last 2 months</option>
+                        <option value="custom">Custom</option>
                     </select>
+                    <br>
+                    <div id="custom_last_match" class='custom_needle_range'>
+                        <select id="sel_custom_last_match" class="form-control custom_select">
+                            <option value='min'>Last</option>
+                            <option value='max'>Not last</option>
+                        </select>
+                        %= datetime_field 'text_custom_last_match', (id => 'last_match_date') => (size => 14) => (class => 'form-control custom_datetime') => (value => $deftime)
+                        <input type="button" id="btn_custom_last_match" class="btn" value="Apply">
+                    </div>
```

Comment:
Instead of duplicating for "seen" and "match" you can try to define a helper, e.g. in https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/WebAPI/Plugin/Helpers.pm and use that in both places with "seen" and "match" as arguments

Reply:
Do you mean I could define a helper including all seen and match contents, e.g all code in `<form>`?

Reply:
Writing a custom helper would likely mean making all these elements via Perl code. Including a nested template is likely the better alternative here.

Reply:
I changed that to use a nested template.